cmd/self-update.go

```diff
@@ -27,17 +29,22 @@ func init() {
 func NewKoolSelfUpdate() *KoolSelfUpdate {
 	return &KoolSelfUpdate{
 		*newDefaultKoolService(),
+		task.NewRunner(),
 		&updater.DefaultUpdater{RootCommand: rootCmd},
 	}
 }
 
 // Execute runs the self-update logic with incoming arguments.
 func (s *KoolSelfUpdate) Execute(args []string) (err error) {
-	var latestVersion semver.Version
+	var currentVersion, latestVersion semver.Version
 
-	currentVersion := s.updater.GetCurrentVersion()
+	err = s.taskRunner.Run("Updating kool version", func() (taskError error) {
```

Comment:
I understand the practice this gives  us, but I am a bit worried there is a hidden race condition where we don't know for sure if the closure was ever executed, but we do conditionally behave below on something that depends on it. Maybe I would suggest getting all the behaviour below and bring it to within this closure.